internal/service/batch/job_definition.go

```diff
@@ -782,6 +782,7 @@ func resourceJobDefinitionUpdate(ctx context.Context, d *schema.ResourceData, me
 		jobDefinitionType := awstypes.JobDefinitionType(d.Get(names.AttrType).(string))
 		input := &batch.RegisterJobDefinitionInput{
 			JobDefinitionName: aws.String(name),
+			Tags:              getTagsIn(ctx),
```

Comment:
Because the create, read, and update operations all have a `Tags` field, the `identifierAttribute` can be removed from the `// @Tags` annotation.

This will remove an unnecessary tag update before the update operation

Reply:
I'm going to keep because if only `tags` change then we _do_ want to call `UpdateTags`.